pages/detention-locations.vue

```diff
@@ -0,0 +1,228 @@
+<template>
+  <loading-message v-if="loading"/>
+  <map-component v-else
+                 :pageInfo="pageInfo"                
+                 :originalMarkers="originalMarkers"
+                 :filtersGeneralInfo="filtersGeneralInfo"
+                 :filterProperties="filterProperties"
+                 iconBehaviour="changing"
+                 :includeTimeline="false">
+  </map-component>
+</template>
+
+<script>
+
+ import {SPARQLQueryDispatcher} from '~/assets/js/SPARQLQueryDispatcher';
+ import APIDataHandler from '~/assets/js/APIDataHandler';
+ import json from '../big-query-output.json';
+ import FilteringMethods from '../assets/js/FilteringMethods';
+ import MapComponent from '../components/MapComponent.vue';
+ import LoadingMessage from '../components/LoadingMessage.vue';
+ import Swal from "sweetalert2";
+
+ definePageMeta({
+   layout: 'default'
+ })
+
+ export default {
+   components: { MapComponent, LoadingMessage },
+   data: () => ({
+     pageInfo: {
+       title: 'Places of Detention for Accused Witches',
+       html: '<div>This map shows the different locations for accused witches place of detention. These people were imprisoned either before or after their witchcraft trial and could be <b class="font-bold">imprisoned multiple times and in different locations</b>. This dataset has records for only <b class="font-bold">560</b> instances of imprisonment for the <b class="font-bold">3212</b> named witches. Again, for most of the accused witches, the surviving documentation does not provide information about their imprisonment.</div>',
+       footer: 'witches.is.ed.ac.uk',
+       confirmButtonText: 'Close',
+       type: 'info',
+       showCloseButton: true,
+     },
+     queryOutput: json,
+     sparqlUrl: 'https://query.wikidata.org/sparql',
+     wikiPages: [],
+     loading: true,
+     originalMarkers: [],
+     filtersToFind: [
+       ["socialClass", "changing"],
+       ["occupation", "changing"],
+     ],
+     filtersGeneralInfo: {
+       title: "Accused witch filters",
+       filtersShowing: true
+     },
+     filterProperties: {
+       sex: {
+         label: "Gender",
+         description: "84% of the accused were women and 15% men. The sex is not known for 1% of those accused.  A gender division of 85% women and 15% men is seen in most other parts of Europe, but in areas like Estonia, Russia and Finland the percentage of men accused is as high and in some areas higher than of women. In Iceland the percentage of men executed was as high as 90%. ",
```

Comment:
noticing there is a lot of repetition with the filter properties. It seems the same filter text is repeated in other files. It is possible to extract this so if we need to edit it, we only have to do it once rather than in three different locations?

Reply:
good idea, have created a json file and read in from it and matched with the key in mounted. Should I make the same changes for the other pages in a new branch? if so a branch of this branch or of the branch with all the pages added that hasn't been reviewed yet?

Reply:
Thanks Ruby. As noted, if we merge this in, we can then merge the parent branch into all the other branches remaining and update them accordingly.